AX: Expose orientation on iOS

hoffmanjoshua opened the pull request:
#### 44a63befef09f0f6755ed771051826780cde1fb2
<pre>
AX: Expose orientation on iOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=299043">https://bugs.webkit.org/show_bug.cgi?id=299043</a>
<a href="https://rdar.apple.com/160805653">rdar://160805653</a>

Reviewed by Chris Fleizach.

Begin exposing the accessibility orientation of objects (when supported) on iOS. This
PR explicitly only will expose an non-undefined orientation when the object (a) supports
aria-orientation and (b) has a non-default value.

* Source/WebCore/accessibility/AXCoreObject.cpp:
(WebCore::AXCoreObject::defaultOrientation const):
(WebCore::AXCoreObject::orientation const):
* Source/WebCore/accessibility/AXCoreObject.h:
* Source/WebCore/accessibility/ios/WebAccessibilityObjectWrapperIOS.mm:
(-[WebAccessibilityObjectWrapper accessibilityOrientation]):

Canonical link: <a href="https://commits.webkit.org/300166@main">https://commits.webkit.org/300166@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/29b023c7fbe77c2fada1a6437d8fbf763e62711c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/121477 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/41174 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/31833 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/127927 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/73560 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/4c133af7-aed6-4fa7-9e33-5f65b32463f3) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/123353 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/41876 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/49753 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/92262 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61383 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/62b3e12a-a1ca-44dd-9e35-ae55f34894d6) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/124429 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33435 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/108814 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/72939 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/30e5c8b9-7c6a-4782-9473-4f6e4b4e3721) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32444 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/26990 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/71498 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/102921 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/27150 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/130752 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/48405 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/36802 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/100852 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/48773 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/105035 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/100759 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25565 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46168 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/24246 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/45101 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/48264 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/53976 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/47735 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/51081 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/49417 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

minorninth commented on Source/WebCore/accessibility/AXCoreObject.cpp:
```diff
@@ -932,6 +932,30 @@ bool AXCoreObject::selfOrAncestorLinkHasPopup() const
+    case AccessibilityRole::List:
+    case AccessibilityRole::ListBox:
+    case AccessibilityRole::Menu:
+    case AccessibilityRole::ScrollBar:
```
I lean towards Unknown for this since both horizontal and vertical are common

minorninth reviewed (approved):
Great!

hoffmanjoshua committed:
AX: Expose orientation on iOS
https://bugs.webkit.org/show_bug.cgi?id=299043
rdar://160805653

Reviewed by Chris Fleizach.

Begin exposing the accessibility orientation of objects (when supported) on iOS. This
PR explicitly only will expose an non-undefined orientation when the object (a) supports
aria-orientation and (b) has a non-default value.

* Source/WebCore/accessibility/AXCoreObject.cpp:
(WebCore::AXCoreObject::defaultOrientation const):
(WebCore::AXCoreObject::orientation const):
* Source/WebCore/accessibility/AXCoreObject.h:
* Source/WebCore/accessibility/ios/WebAccessibilityObjectWrapperIOS.mm:
(-[WebAccessibilityObjectWrapper accessibilityOrientation]):

Canonical link: https://commits.webkit.org/300166@main
